frontend/src/metabase/querying/parameters/utils/normalize.ts

```diff
@@ -0,0 +1,58 @@
+import type { DateFilterValue } from "metabase/querying/filters/types";
+import type { ParameterValueOrArray } from "metabase-types/api";
+
+import { deserializeDateFilter } from "./dates";
+
+function normalizeArray(value: ParameterValueOrArray | null) {
+  if (value == null) {
+    return [];
+  } else {
+    return Array.isArray(value) ? value : [value];
+  }
+}
+
+export function normalizeStringParameterValue(
```

Comment:
These functions would be used in refactored parameter widgets; there would be a single source of truth how to make sense of the saved parameter value.